src/box/lua/load_cfg.lua

```diff
@@ -662,43 +699,58 @@ local function compare_cfg(cfg1, cfg2)
     return table.equals(cfg1, cfg2)
 end
 
+local function try_rollback_option(oldcfg, key)
```

Comment:
I don't like this function, because it requires for the option to be revertable (have an entry in the `dynamic_cfg_revert`). I'd rather keep it inline or introduce a different function instead, which would revert the option to the old value and call the 'revert' callback if it's available. Also, I'd put the revert function in the `config_modules` table.